### PR TITLE
Parametrize Binomial and Categorical distributions via logit_p

### LINKED ISSUE #5005: Parametrize Binomial and Categorical distributions via logit_p
## Description of your problem

`Bernoulli` can take `logit_p` as an argument:

https://github.com/pymc-devs/pymc3/blob/59ce9af97f0b840f992322f33ec8dee203d7d1c0/pymc3/distributions/discrete.py#L334-L339

Could `Binomial` also take this argument?

### PR DESCRIPTION
**Pull request by neelriyer**
**What are the (breaking) changes that this PR makes?**
This PR parametrizes the Binomial and Categorical distributions via `logit_p`
 
**Important background, or details about the implementation**
Resolves #5005
 
**[linting/style checks have been run](https://github.com/pymc-devs/pymc3/wiki/PyMC3-Python-Code-Style)**
Yes

### CODE REVIEW

**Comment by michaelosthege** [comment]
@spiyer99 it looks like tests are still needed

**Comment by MarcoGorelli** [comment]
Agreed about tests - might also be worth making a helper function for this, if it's being reused in a few different places

**Comment by ricardoV94** [comment]
There are already tests with reparametrizations in `test_distributions.py` and `test_distrubitions_random.py`, that can be used as a template

**Comment by ricardoV94** [comment]
This might also be helpful: https://github.com/pymc-devs/pymc3/blob/main/docs/source/developer_guide_implementing_distribution.md

**Comment by neelriyer** [comment]
Sorry for the late reply and thanks for the help guys. I've been super busy. I'll add in the tests when I get some time this week.

**Comment by neelriyer** [comment]
@ricardoV94 could I add my tests to the `test_binomial` function in the `test_distributions.py` file? What do you think of this approach?

**Comment by michaelosthege** [comment]
@spiyer99 I recommend to create a new test method, similar to [this one](https://github.com/pymc-devs/pymc3/blob/c6e9153d7ab36055aaafe778abd168741555e4af/pymc/tests/test_distributions.py#L1312-L1330), specifically for testing alternative parametrizations.

**Comment by ricardoV94** [comment]
> @ricardoV94 could I add my tests to the `test_binomial` function in the `test_distributions.py` file? What do you think of this approach?

test_binomial makes sense only for logp/logcdf tests. 

If you want to validate the parameters are correctly converted you should probably check the tests in `test_distributions_random.py` such as `TestNormalTau`. That's also where tests like the one @michaelosthege suggested should probably go.

**Comment by ricardoV94** [comment]
I think for the Categorical we want a softmax transformation, not just a logit one

**Comment by neelriyer** [comment]
> If you want to validate the parameters are correctly converted you should probably check the tests in test_distributions_random.py such as TestNormalTau. That's also where tests like the one @michaelosthege suggested should probably go.

@ricardoV94, I added tests to validate that the parameters are correctly converted using `TestNormaltau` as a template. 
I also added the tests that @michaelosthege suggested. These are included in the `test_random_distributions.py` file. 
Not sure if I’ve done this correctly. Let me know.

> I think for the Categorical we want a softmax transformation, not just a logit one

Is there an example in the code that I can use to understand how to apply the softmax transformation?

**Comment by ricardoV94** [comment]
> Is there an example in the code that I can use to understand how to apply the softmax transformation?

something like this might work
```python
if (p is None and logit_p is None) or (p is not None and logit_p is not None):
  raise ValueError(...)

elif logit_p is not None:
  p = pm.math.softmax(logit_p, axis=-1)
```

**Comment by ricardoV94** [comment]
Closing this due to inactivity. Feel free to reopen a new PR for this anytime
